### PR TITLE
Initial PR for version 1.2

### PR DESCRIPTION
**Pull request by jarmokortetjarvi**

### CODE REVIEW

**Comment by mlaitinen** [comment]
Excellent! I'm very happy to see Finnish contributions.

We have been working on similar module https://github.com/avoinsystems/l10n_fi_rapko which is based on [Raportointikoodisto](http://www.raportointikoodisto.fi/). I think we should work together to integrate both of the accounting charts to this repository so that they'd share common data (ie. taxes, fiscal positions).

**Comment by jarmokortetjarvi** [comment]
I absolutely agree about working together to build one comprehensive Finnish localization module.

I have examined the repository you linked earlier with some detail, having the integration of both implementations in mind. It pains me to know that the community would already have a very mature l10n_fi, if this version was released in the beginning of the year, when you have worked on your module.

**Comment by jarmokortetjarvi** [comment]
The main "problem" with the version in this PR is that it uses the full Liikekirjurit chart of accounts. It could be a good idea to make the main version with less accounts (e.g. http://www.tilit.info/tase_suppea.php and http://www.tilit.info/tuloslaskelma_suppea.php). 

My idea would be to do that by trimming the current chart, preserving all the ids, so it will be possible to upgrade the minimal "default" chart of accounts to the full version later without any conflicts.

**Comment by dreispt** [comment]
Thanks for contributing.
The module should be contained in a subdirectory (`l10n_fi_account/` maybe?) and not be at the root of the repository.

**Comment by mlaitinen** [comment]
Sorry for bringing this up until now, but I agree with @dreispt. If the repository is called `l10n_fi` then maybe we should put directories like 
- `l10n_fi_account` or `l10n_fi_liikekirjuri` (for @dreispt: liikekirjuri is a common chart of accounts in Finland)
- `l10n_fi_rapko` for Raportointikoodisto (another chart of accounts with its structure designed for official reports)
- `l10n_fi_invoice` for an invoice tempate that fills the Finnish legal requirements for invoices (this is something we're releasing hopefully this week)

IMHO if there's going to be a module called `l10n_fi`, it should contain everything that is essential for every Finnish company using Odoo, eg. address format, payment reference on the invoice, and so on. Or maybe that's something we should contribute directly to Odoo S.A. (?)

**Comment by jarmokortetjarvi** [comment]
No problem. This is a great stage to bring out observations and opinions. I actually agree with your suggestion.

I think l10n_fi_account is a better name if we plan to combine the two accounting charts that we currently have. The actual combining, however, may not be possible due different basic structures in the charts.

In other case l10n_fi_liikekirjuri would be a better name.

However, I think we should steer towards one chart of accounts. I grudgingly think it might be better to base it to raportointikoodisto to support it becoming the standard chart of accounts.

File: l10n_fi_liikekirjuri/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+##############################################################################
+{
+    'name': 'Finland - Accounting',
+    'category': 'Localization/Account Charts',
+    'version': '1.2',
+    'author': 'Vizucom Oy, Oy Tawasta OS Technologies Ltd., Odoo Community Association (OCA)',
+    'license': 'AGPL-3',
+    'website': 'http://www.vizucom.com, http://www.tawasta.com',
+    'depends': ['sale_stock', 'report_webkit', 'account'],
+    'description': """
```
Comment (dreispt) [inline l10n_fi_liikekirjuri/__openerp__.py]: You can remove this for the README.rst to be used instead.

File: l10n_fi_liikekirjuri/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+Based on 'Liikekirjurit' accounts chart version 7.0
+(http://www.kirjurituote.fi/tuotteet/liikekirjuri/)
+
+This is the comprehensive version of the accounts chart,
+and should be sufficient for all industries and company types.
+
+The number of accounts is approx. 950
+The number of headers is approx. 300
+""",
+    'depends': ['base_iban', 'base_vat', 'account_chart'],
```
Comment (dreispt) [inline l10n_fi_liikekirjuri/__openerp__.py]: You have two `depends`keys!

File: l10n_fi_liikekirjuri/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    'data': [
+        'data/account.account.type.csv',
+        'data/account.account.template.csv',
+        'data/account.tax.code.template.csv',
+        'data/account.chart.template.csv',
+        'data/account.tax.template.csv',
+    ],
+    'installable': 'True',
+    'images': [],
```
Comment (dreispt) [inline l10n_fi_liikekirjuri/__openerp__.py]: deprecated, remove

**Comment by dreispt** [comment]
Please remove pyc files.
You should add repo initialization files, such as `.travis.yml`. 
See https://github.com/OCA/maintainer-quality-tools/tree/master/sample_files

File: l10n_fi_liikekirjuri/.gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+.*
```
Comment (dreispt) [inline l10n_fi_liikekirjuri/.gitignore]: Move gitignore to the root.
You may prefer to use the one at https://github.com/OCA/maintainer-quality-tools/tree/master/sample_files

File: l10n_fi_liikekirjuri/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    'name': 'Finland - Accounting',
+    'category': 'Localization/Account Charts',
+    'version': '1.2',
```
Comment (dreispt) [inline l10n_fi_liikekirjuri/__openerp__.py]: 8.0.1.2.0
(see https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers)

**Comment by dreispt** [comment]
We need confirmation for the CLA.

**Comment by jarmokortetjarvi** [comment]
My CLA was confirmed on 26.08.2015

**Comment by dreispt** [comment]
Manually verified CLAs are posted on the thread by someone from the board. Can you ask the person that checked it to confirm here?

**Comment by dreispt** [comment]
:+1: Looks good! Just need some confirmation of the CLA check (and a second review)

**Comment by mlaitinen** [comment]
I have already installed this module in my own environment but unfortunately haven't had the time to review it. I'm trying to get it done this week.

**Comment by mlaitinen** [comment]
There are some account names with whitespaces in them, such as "V a s t a t t a v a a" and "O s a k e y h t i ö", which is something you can't find using the Odoo search, unless you know exactly what you are looking for.

The build seems to be failing with the following error:

```
2015-09-01 10:05:03,953 22634 INFO 3114148-1-c3e0b2-all openerp.modules.module: module l10n_fi_liikekirjuri: creating or updating database tables
2015-09-01 10:05:03,953 22634 WARNING 3114148-1-c3e0b2-all openerp.addons.base.module.module: module l10n_fi_liikekirjuri: description is empty !
2015-09-01 10:05:03,953 22634 INFO 3114148-1-c3e0b2-all openerp.modules.loading: module l10n_fi_liikekirjuri: loading data/account.account.type.csv
2015-09-01 10:05:04,080 22634 INFO 3114148-1-c3e0b2-all openerp.modules.loading: module l10n_fi_liikekirjuri: loading data/account.account.template.csv
2015-09-01 10:05:09,601 22634 INFO 3114148-1-c3e0b2-all openerp.modules.loading: module l10n_fi_liikekirjuri: loading data/account.tax.code.template.csv
2015-09-01 10:05:09,603 22634 ERROR 3114148-1-c3e0b2-all openerp: Failed to initialize database `3114148-1-c3e0b2-all`.
Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/cli/server.py", line 97, in preload_registry
    db, registry = openerp.pooler.get_db_and_pool(dbname,update_module=update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/pooler.py", line 33, in get_db_and_pool
    registry = RegistryManager.get(db_name, force_demo, status, update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/modules/registry.py", line 203, in get
    update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/modules/registry.py", line 233, in new
    openerp.modules.load_modules(registry.db, force_demo, status, update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/modules/loading.py", line 354, in load_modules
    loaded_modules, update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/modules/loading.py", line 256, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/modules/loading.py", line 188, in load_module_graph
    load_data(module_name, idref, mode)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/modules/loading.py", line 76, in <lambda>
    load_data = lambda *args: _load_data(cr, *args, kind='data')
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/modules/loading.py", line 118, in _load_data
    tools.convert_csv_import(cr, module_name, pathname, fp.read(), idref, mode, noupdate)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/tools/convert.py", line 935, in convert_csv_import
    raise Exception(_('Module loading %s failed: file %s could not be processed:\n %s') % (module, fname, warning_msg))
Exception: Module loading l10n_fi_liikekirjuri failed: file l10n_fi_liikekirjuri/data/account.tax.code.template.csv could not be processed:
 Line 1 : sequence
```

Getting closer .. :)

**Comment by dreispt** [comment]
@jarmokortetjarvi It would help reviews if you added the `.travis.yml` file at the root as I mentioned before.

**Comment by mlaitinen** [comment]
There's something horribly wrong with the Odoo Runbot. I just rebuilt the latest commit and I got logged in to OpenERP v7 ??

**Comment by gurneyalex** [comment]
Hi
I noticed the branches 8.0 and 9.0 were not there in this repository, so I created them. Resubmitting this PR with the correct  target branch should allow runbot to build correctly.

**Pull request closed without merging**